timing/cu/vectormemoryunit.go

```diff
@@ -109,8 +109,9 @@ func (u *VectorMemoryUnit) insertTransactionToPipeline(
 }
 
 func (u *VectorMemoryUnit) execute(now sim.VTimeInSec) (madeProgress bool) {
-	item := u.postInstructionPipelineBuffer.Pop()
+	item := u.postInstructionPipelineBuffer.Peek()
 	if item == nil {
+		u.postInstructionPipelineBuffer.Pop()
```

Comment:
This line is actually not needed. If there is no item in the buffer, pop will not do anything.